### PR TITLE
Improve json hpp install

### PR DESCRIPTION
**Pull request by NicolasToussaint**
## Description

I have several difficulties when installing Fossology from sources:
- sometimes my network is unstable, and the whole install process fails because the download of the json.hpp file fails
- Quite often, the machines I install Fossology on require a proxy to access the internet.

### Changes

- Check first that file final file is not already available (sith correct checksum)
- Added the Retry and Timeout options to curl, to make the installation process more robust
- Let the user configure a proxy by sourcing a new configuration file: `/etc/fossology/fossology-proxy.conf`

## How to test

Install Fossology from source, through a proxy

### CODE REVIEW

File: install/scripts/install-json-hpp.sh [hunk 1]
```diff
@@ -2,20 +2,42 @@
 
 cd "$(dirname "$0")/../.."
 FINAL_FILE=./src/copyright/agent/json.hpp
-SHASUM=fbdfec4b4cf63b3b565d09f87e6c3c183bdd45c5be1864d3fcb338f6f02c1733
-TMP=$(mktemp)
-CHECKSUMFILE=$(mktemp)
+FILE_SHASUM="faa2321beb1aa7416d035e7417fcfa59692ac3d8c202728f9bcc302e2d558f57"
+FILE_URL="https://github.com/nlohmann/json/releases/download/v2.1.1/json.hpp"
```
[the remark this answers is not in the document]
Comment (NicolasToussaint) [reply install/scripts/install-json-hpp.sh]: Oh, thanks, will fix that.

File: install/scripts/install-json-hpp.sh [hunk 2]
```diff
@@ -2,20 +2,42 @@
...
+}
+
+# Check if file is already available
+_check_file "$SHASUM" "$FINAL_FILE" && exit 0
 
-curl -s -o "$TMP" -L https://github.com/nlohmann/json/releases/download/v3.1.2/json.hpp
-echo "$SHASUM $TMP" > "$CHECKSUMFILE"
-sha256sum --quiet -c "$CHECKSUMFILE"
+# Proxy Setup: If needed, configure the proxy variables in the appropriate config file
+proxy_file=/etc/fossology/fossology-proxy.conf
```
[the remark this answers is not in the document]
Comment (NicolasToussaint) [reply install/scripts/install-json-hpp.sh]: Thanks, didn't realize I will reuse that config entry

File: install/scripts/install-json-hpp.sh [hunk 3]
```diff
@@ -1,21 +1,43 @@
...
+conf_file=/etc/fossology.conf
+[ -r $conf_file ] && . $conf_file
+
 cd "$(dirname "$0")/../.."
 FINAL_FILE=./src/copyright/agent/json.hpp
-SHASUM=fbdfec4b4cf63b3b565d09f87e6c3c183bdd45c5be1864d3fcb338f6f02c1733
-TMP=$(mktemp)
-CHECKSUMFILE=$(mktemp)
+FILE_SHASUM="fbdfec4b4cf63b3b565d09f87e6c3c183bdd45c5be1864d3fcb338f6f02c1733"
+FILE_URL="https://github.com/nlohmann/json/releases/download/v2.1.1/json.hpp"
```
Comment (GMishx) [inline install/scripts/install-json-hpp.sh]: Please check the version as well. (v2.1.1 > v3.1.2)

File: install/scripts/install-json-hpp.sh [hunk 4]
```diff
@@ -1,21 +1,43 @@
 #!/bin/bash
 
+# Source fossology config file, for Proxy settings
+conf_file=/etc/fossology.conf
```
Comment (GMishx) [inline install/scripts/install-json-hpp.sh]: It is not a good idea to hard code path here.
If you are installing FOSSology from Debian packages, then this path is correct.
If you install FOSSology from source, the path is more like `/usr/local/etc/fossology/fossology.conf`

File: install/scripts/install-json-hpp.sh (hunk 3 above)
Comment (NicolasToussaint) [reply install/scripts/install-json-hpp.sh]: Stupid me, URL is also fixed.

File: install/scripts/install-json-hpp.sh (hunk 4 above)
Comment (GMishx) [reply install/scripts/install-json-hpp.sh]: You can try the process target here:
https://github.com/fossology/fossology/blob/master/Makefile.process#L16
And make something like:
https://github.com/fossology/fossology/blob/master/install/defconf/fossology.conf.in#L48

The variables which you can look into are:
https://github.com/fossology/fossology/blob/master/Makefile.conf#L40

Comment (NicolasToussaint) [reply install/scripts/install-json-hpp.sh]: Many thanks for the explanations and pointers - I have removed the proxy part of the PR altogether, I will have to work on this separately later.

**Comment by ag4ums** [comment]
the fix is here #1226  so closing this.

**Pull request closed without merging**